### ISSUE TITLE
Remove use of deprecated null_data_source

### ISSUE THREAD

**Issue by bwmetcalf** [cloudposse/terraform-aws-ec2-instance#93]
The use of null_data_source is now deprecated and locals should be used:  https://registry.terraform.io/providers/hashicorp/null/latest/docs/data-sources/data_source.  Using null_data_source results in the following warning:
```
Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals
```

**Comment by bwmetcalf** [comment]
https://github.com/cloudposse/terraform-aws-ec2-instance/pull/94

**Comment by Gowiem** [comment]
Closed via #94
